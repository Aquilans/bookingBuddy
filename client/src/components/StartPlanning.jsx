import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';

// Components
import TripCreationForm from './TripCreationForm.jsx';
import TripPreferencesForm from './TripPreferencesForm.jsx';

const PlanningLinks = () => (
  <div>
    <Link to="/start-planning/trip-create">Create Trip</Link>
    <Link to="/start-planning/trip-preferences">Trip Preferences</Link>
  </div>
);

const PlanningRoutes = () => (
  <Router>
    <div>
      <PlanningLinks />
      <Route
        path="/start-planning/trip-create"
        component={TripCreationForm} />
      <Route
        path="/start-planning/trip-preferences"
        component={TripPreferencesForm} />
    </div>
  </Router>
);

class StartPlanning extends Component {

  render() {
    return (
      <div>
        <h2>Start Planning</h2>
        <PlanningRoutes />
      </div>
    );
  }
}

export default StartPlanning;
