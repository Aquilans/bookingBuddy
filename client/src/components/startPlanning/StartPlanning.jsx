import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';

// Components
import TripCreationForm from './startPlanningComponents/TripCreationForm.jsx';
import TripPreferencesForm from './startPlanningComponents/TripPreferencesForm.jsx';

const PlanningLinks = () => (
  <div>
    <Link className="orange btn" to="/start-planning/trip-create">Create Trip</Link>
    <Link className="orange btn" to="/start-planning/trip-preferences">Trip Preferences</Link>
  </div>
);

const PlanningRoutes = ({userEmail, tripId, profile}) => (
  <Router>
    <div>
      <PlanningLinks />
      <Route
        path="/start-planning/trip-create"
        render={() => <TripCreationForm userEmail={userEmail} /> } />
      <Route
        path="/start-planning/trip-preferences"
        render={() => <TripPreferencesForm profile={profile} userEmail={userEmail} tripId={tripId} /> } />
    </div>
  </Router>
);

const StartPlanning = ({userEmail, tripId, profile}) => (
  <div className="container start-planning">
    <h2>Start Planning</h2>
    <PlanningRoutes profile={profile} userEmail={userEmail} tripId={tripId} />
  </div>
);

export default StartPlanning;
