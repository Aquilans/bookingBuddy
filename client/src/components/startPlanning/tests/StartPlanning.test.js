import React from 'react';
import renderer from 'react-test-renderer';

import StartPlanning from '../StartPlanning.jsx';

it('renders correctly', () => {
  const startPlanning = renderer.create(
    <StartPlanning />
  ).toJSON();
  expect(startPlanning).toMatchSnapshot();
});