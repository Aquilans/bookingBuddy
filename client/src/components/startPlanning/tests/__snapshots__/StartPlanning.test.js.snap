// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <h2>
    Start Planning
  </h2>
  <div>
    <div>
      <a
        className="waves-effect waves-light orange btn"
        href="/start-planning/trip-create"
        onClick={[Function]}
      >
        Create Trip
      </a>
      <a
        className="waves-effect waves-light orange btn"
        href="/start-planning/trip-preferences"
        onClick={[Function]}
      >
        Trip Preferences
      </a>
    </div>
  </div>
</div>
`;
